GitCommands/Git/GitSubmoduleManager.cs

```diff
@@ -0,0 +1,101 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using GitCommands.Config;
+using GitUIPluginInterfaces;
+
+namespace GitCommands.Git
+{
+    public interface IGitSubmoduleManager
```

Comment:
I have a mild-to-strong aversion to classes named 'Manager'.

https://blog.codinghorror.com/i-shall-call-it-somethingmanager/
https://stackoverflow.com/questions/1866794/naming-classes-how-to-avoid-calling-everything-a-whatevermanager

Let's dig deeper into what these types should be doing and name them such that their responsibilities are more self-obvious.

Reply:
No arguments at all. I don't have a big love for "Manager"s either and I loath "Helper" classes.
It was a quick POC late at night and my vocabulary was getting limited.